docs/usage/design.rst

```diff
@@ -0,0 +1,26 @@
+Design of SCAutolib
+=============================
+
+This section would introduce design decisions that were made during
+implementation with explanation why is it decided in that particular way.
+
+* **SCAutolib uses MVC pattern adopted for CLI and test usage.**
+
+  In earlier versions of the SCAutolib (those, that are not even released) there
+  was basically no design defined. To clarify this situation and to have some
+  base for better design we choose MVC pattern, where:
+
+  * Models are all basic classes in the library (CAs, Users, Card, etc.) that
+    provides low-level atomic functionality
+  * View is a CLI interface
+  * Controller is a heart of the library that handles all (two) variants of the
```

Comment:
- use 'both' instead 'all (two)'
- I would also write something like "...library that handles logic behind both variants of..."